source/views/settings/screens/overview/developer.tsx

```diff
@@ -32,24 +47,73 @@ export class DeveloperSection extends React.Component<Props> {
 		}
 	}
 
-	render() {
-		return (
+	const checkForValidity = (text: string) => {
+		const pattern = /^(http|https):\/\/[^ "]+$/u
+		const isUrlValid = pattern.test(text)
+		const isValid = isUrlValid || text.length === 0
+
+		isValid ? setErrorMessage('') : setErrorMessage('Waiting for a valid url…')
+	}
+
+	const handleOnChange = (text: string) => {
+		setServerAddress(text)
+		checkForValidity(text)
+	}
+
+	const refreshApp = () => {
+		storage.setServerAddress(serverAddress)
+		restart.Restart()
+	}
+
+	React.useMemo(async () => {
+		if (!serverAddress && !_serverAddress.current) {
+			const address = await storage.getServerAddress()
+			setServerAddress(address)
+		}
+	}, [serverAddress])
+
+	return (
+		<>
 			<Section header="DEVELOPER">
-				<PushButtonCell onPress={this.onAPIButton} title="API Tester" />
+				<PushButtonCell onPress={onAPIButton} title="API Tester" />
+				<PushButtonCell onPress={onBonAppButton} title="Bon Appetit Picker" />
+				<PushButtonCell onPress={onDebugButton} title="Debug" />
 				<PushButtonCell
-					onPress={this.onBonAppButton}
-					title="Bon Appetit Picker"
-				/>
-				<PushButtonCell onPress={this.onDebugButton} title="Debug" />
-				<PushButtonCell
-					onPress={this.sendSentryMessage}
+					onPress={sendSentryMessage}
 					title="Send a Sentry Message"
 				/>
 				<PushButtonCell
-					onPress={this.sendSentryException}
+					onPress={sendSentryException}
 					title="Send a Sentry Exception"
 				/>
 			</Section>
-		)
-	}
+
+			<Section
```

Comment:
On the other hand… does it make it significantly cleaner to make a ServerUrlSection component?

Reply:
It might make more sense for this to be pulled out. I'll see what I can do!